### PR TITLE
Fix and improve TLS documentation

### PR DESCRIPTION
**Pull request by jrtc27**
The existing calculations in the table didn't make any sense. We now also define that we're a version of Variant I and how to set up tp.

Closes: #53

### CODE REVIEW

**Comment by jrtc27** [comment]
@MaskRay can't add you as a reviewer but please review this :)

File: riscv-elf.md [hunk 1]
```diff
@@ -523,10 +523,10 @@ Enum | ELF Reloc Type        | Description                     | Field       | C
...
 6    | R_RISCV_TLS_DTPMOD32  | TLS relocation                  | _word32_    | S->TLSINDEX
 7    | R_RISCV_TLS_DTPMOD64  | TLS relocation                  | _word64_    | S->TLSINDEX
-8    | R_RISCV_TLS_DTPREL32  | TLS relocation                  | _word32_    | S + A + TLS - TLS_TP_OFFSET
-9    | R_RISCV_TLS_DTPREL64  | TLS relocation                  | _word64_    | S + A + TLS - TLS_TP_OFFSET
-10   | R_RISCV_TLS_TPREL32   | TLS relocation                  | _word32_    | S + A + TLS + S_TLS_OFFSET - TLS_DTV_OFFSET
-11   | R_RISCV_TLS_TPREL64   | TLS relocation                  | _word64_    | S + A + TLS + S_TLS_OFFSET - TLS_DTV_OFFSET
+8    | R_RISCV_TLS_DTPREL32  | TLS relocation                  | _word32_    | S + A - TLS_DTV_OFFSET
+9    | R_RISCV_TLS_DTPREL64  | TLS relocation                  | _word64_    | S + A - TLS_DTV_OFFSET
+10   | R_RISCV_TLS_TPREL32   | TLS relocation                  | _word32_    | S + A + TLS
+11   | R_RISCV_TLS_TPREL64   | TLS relocation                  | _word64_    | S + A + TLS
```
Comment (aswaterman) [inline riscv-elf.md:529]: I'm guessing by "+ TLS" we mean "+ TCB", right?  Or is this the standard syntax other psABIs use to indicate that concept?

Comment (jrtc27) [reply riscv-elf.md:529]: I interpret it to mean "TLS offset assigned to the object's static block", i.e. what the ELF TLS "spec" (and FreeBSD) calls "tlsoffset_N"

Comment (jrtc27) [reply riscv-elf.md]: Does my updated version make more sense?

Comment (aswaterman) [reply riscv-elf.md]: Yeah, this version's clearer to me.

File: riscv-elf.md [hunk 2]
```diff
@@ -521,12 +521,12 @@ Enum | ELF Reloc Type        | Description                     | Field       | C
...
-7    | R_RISCV_TLS_DTPMOD64  | TLS relocation                  | _word64_    | S->TLSINDEX
-8    | R_RISCV_TLS_DTPREL32  | TLS relocation                  | _word32_    | S + A + TLS - TLS_TP_OFFSET
-9    | R_RISCV_TLS_DTPREL64  | TLS relocation                  | _word64_    | S + A + TLS - TLS_TP_OFFSET
-10   | R_RISCV_TLS_TPREL32   | TLS relocation                  | _word32_    | S + A + TLS + S_TLS_OFFSET - TLS_DTV_OFFSET
-11   | R_RISCV_TLS_TPREL64   | TLS relocation                  | _word64_    | S + A + TLS + S_TLS_OFFSET - TLS_DTV_OFFSET
+6    | R_RISCV_TLS_DTPMOD32  | TLS relocation                  | _word32_    | TLSINDEX
+7    | R_RISCV_TLS_DTPMOD64  | TLS relocation                  | _word64_    | TLSINDEX
+8    | R_RISCV_TLS_DTPREL32  | TLS relocation                  | _word32_    | S + A - TLS_DTV_OFFSET
+9    | R_RISCV_TLS_DTPREL64  | TLS relocation                  | _word64_    | S + A - TLS_DTV_OFFSET
+10   | R_RISCV_TLS_TPREL32   | TLS relocation                  | _word32_    | S + A + TLSOFFSET
```
Comment (MaskRay) [inline riscv-elf.md:528]: Probably worth clarifying that `R_RISCV_TLS_TPREL32` and `R_RISCV_TLS_TPREL64` are dynamic relocation types used in the initial exec TLS model.

File: riscv-elf.md [hunk 3]
```diff
@@ -585,15 +585,17 @@ assembler syntax, including that of the relocations, is described in the
...
+Variable  | Description
+:-------  | :----------------
+A         | Addend field in the relocation entry associated with the symbol
+B         | Base address of a shared object loaded into memory
+G         | Offset of the symbol into the GOT (Global Offset Table)
+P         | Position of the relocation
+S         | Value of the symbol in the symbol table
+V         | Value at the position of the relocation
+GP        | Value of `__global_pointer$` symbol
+TLSINDEX  | TLS module index for the object containing the symbol
```
Comment (MaskRay) [inline riscv-elf.md:597]: In the ELF TLS doc, `tls_index` refers to the structure containing a module ID and an offset.
So it is probably better avoiding TLSINDEX.

**Review by MaskRay — APPROVED** [review]
Sure:)

This matches my understanding: https://maskray.me/blog/2021-02-14-all-about-thread-local-storage

**Review by jim-wilson — APPROVED** [review]

File: riscv-elf.md (hunk 3 above)
Comment (jrtc27) [reply riscv-elf.md]: Now using TLSMODULE to mirror tls_index's ti_module

File: riscv-elf.md (hunk 2 above)
Comment (jrtc27) [reply riscv-elf.md]: Added "Runtime" to the start of the description

**Review by kito-cheng — APPROVED** [review]

**Pull request merged**